Assets/Scripts/Managers/AudioManager.cs

```diff
@@ -0,0 +1,77 @@
+using UnityEngine.Audio;
+using System;
+using UnityEngine;
+
+
+public class AudioManager : Singleton<AudioManager>
+{
+    public static AudioManager Instance;
+    public AudioMixerGroup musicMixerGroup;
+    public AudioMixerGroup sfxMixerGroup;
+    public Sound[] sounds;
+
+    void Awake()
+    {
+
+        Setup();
+    }
+
+    private void Setup()
+    {
+        foreach (Sound s in sounds)
+        {
+            s.source = gameObject.AddComponent<AudioSource>();
+            s.Setup();
+            
+            switch (s.audioType)
+            {
+                case Sound.AudioType.music:
+                    s.source.outputAudioMixerGroup = musicMixerGroup;
+                    break;
+
+                case Sound.AudioType.soundEffect:
+                    s.source.outputAudioMixerGroup = sfxMixerGroup;
+                    break;
+            }
+        }
+    }
+    // Update is called once per frame
+    public void Play(string name)
+    {
+        Sound s = System.Array.Find(sounds, sound => sound.name == name);
+        if (s == null) 
+        {
+            Debug.LogWarning("Sound: " + name + " not found!");
+            return;
+        }
+        s.source.Play();
+    }
```

Comment:
Are there any possibilities of more than one sound effect having the same name, and if so, what are the checks to ensure this doesn't happen? Similar question applies to `Stop()` and `Pause()`

Reply:
If no two sounds will have the same name, then I would recommend turning the `sounds` array into a `Dictionary<string, Sound>` for faster access